test/cloudant.view.test.js

```diff
@@ -0,0 +1,97 @@
+// Copyright IBM Corp. 2016. All Rights Reserved.
+// Node module: loopback-connector-cloudant
+// This file is licensed under the Artistic License 2.0.
+// License text available at https://opensource.org/licenses/Artistic-2.0
+
+'use strict';
+
+require('./init.js');
+var Cloudant = require('../lib/cloudant');
+var _ = require('lodash');
+var url = require('url');
+var should = require('should');
+var async = require('async');
+var db, TestView, Random, samples;
+
+describe('cloudant view', function() {
+  describe('viewDocs', function(done) {
+    before(function(done) {
+      db = getDataSource();
+      TestView = db.define('TestView', {
+        name: {type: String},
+      }, {forceId: false});
+      Random = db.define('Random', {
+        foo: {type: String},
+      }, {forceId: false});
+      db.on('connected', function() {
+        async.series([createModelTestView, seedData, insertViewDdoc], done);
+      });
+
+      function createModelTestView(cb) {
+        db.automigrate(cb);
+      };
+      function seedData(cb) {
+        samples = [{name: 'John'}, {name: 'Tom'}, {name: 'Zoe'}];
+        var randomSamples = [{foo: 'bar1'}, {foo: 'bar2'}];
+        TestView.create(samples, function(err) {
+          if (err) return cb(err);
+          Random.create(randomSamples, cb);
+        });
+      };
+      function insertViewDdoc(cb) {
+        // The design doc must be stored with double quote,
+        // and map function cannot join by '+'
+        /* eslint-disable */
+        var ddoc = {
+          "_id": "_design/getModel",
+          "views": {
+            "returnModelInstances": {
+              "map": "function(doc) { if(doc.loopback__model__name) { emit(doc.loopback__model__name, doc); }}"
+            }
+          }
+        };
+        /* eslint-enable */
+        // We test everything in a single db, so the view ddoc
+        // and model 'TestView' should be in same db.
+        db.connector.selectModel('TestView').db.insert(ddoc, cb);
+      };
+    });
+    after(function(done) {
+      TestView.destroyAll(function(err) {
+        if (err) return done(err);
+        Random.destroyAll(done);
+      });
+    });
+
+    it('returns result by query a view', function(done) {
+      db.connector.viewDocs('getModel', 'returnModelInstances',
+        function(err, results) {
+          results.total_rows.should.equal(5);
+          results.rows.forEach(isLBModelInstance);
+          done(err);
+
+          function isLBModelInstance(elem) {
+            elem.value.hasOwnProperty('loopback__model__name').
```

Comment:
If the intended use-case for this feature is existing databases with existing design docs, I would advise against using property names like `loopback__model__name` for a couple reasons:
 1. it might make code readers/reviewers think this is a field added by LoopBack
 2. it might make current/future maintainers think this is an official property name

Adding a property to every document in an existing CouchDB database is a very expensive operation. So expensive it might even be considered impossible by some operators. The only reasonable approach would be to have the user provide the map function and/or property name _already_ being used for this purpose. If that isn't considered upfront, it could lead to assumptions and design decisions that will need to be undone later and may require future breaking changes.